test/eth/TransactionManager.spec.js

```diff
@@ -163,3 +163,103 @@ test('lifecycle hooks', async () => {
   const wipe = cdp.wipeDai(1);
   await Promise.all([txMgr.confirm(wipe), mineBlocks(service)]);
 });
+
+test('lifecycle hooks for give', async () => {
+  TestAccountProvider.setIndex(900);
+  const service = buildTestEthereumCdpService({
+    accounts: {
+      default: {
+        type: 'privateKey',
+        privateKey: TestAccountProvider.nextAccount().key
+      }
+    },
+    log: true
+  });
+
+  await service.manager().authenticate();
+  const txMgr = service.get('smartContract').get('transactionManager');
+
+  const makeListener = (label, state) =>
+    jest.fn(tx => {
+      const { contract, method } = tx.metadata;
+      log(`${label}: ${contract}.${method}: ${state}`);
+    });
+
+  const makeHandlers = label => ({
+    pending: makeListener(label, 'pending'),
+    mined: makeListener(label, 'mined')
+  });
+
+  const open = service.openCdp();
+  log('open id:', uniqueId(open));
+
+  const cdp = await open;
+
+  const newCdpAddress = '0x75ac1188e69c815844dd433c2b3ccad1f5a109ff';
+  const give = cdp.give(newCdpAddress);
+
+  log('give id:', uniqueId(give));
+  
+  const giveHandlers = makeHandlers('give');
+
+  txMgr.listen(give, giveHandlers);
+  await give;
+  
+  expect(giveHandlers.pending).toBeCalled();
+  expect(giveHandlers.mined).toBeCalled();
+});
+
+test('lifecycle hooks for bite', async () => {
+  TestAccountProvider.setIndex(900);
+  const service = buildTestEthereumCdpService({
+    accounts: {
+      default: {
+        type: 'privateKey',
+        privateKey: TestAccountProvider.nextAccount().key
+      }
+    },
+    log: true
+  });
+
+  await service.manager().authenticate();
+  const txMgr = service.get('smartContract').get('transactionManager');
+  const priceService = service.get('price');
+
+  const makeListener = (label, state) =>
+    jest.fn(tx => {
+      const { contract, method } = tx.metadata;
+      log(`${label}: ${contract}.${method}: ${state}`);
+    });
+
+  const makeHandlers = label => ({
+    pending: makeListener(label, 'pending'),
+    mined: makeListener(label, 'mined')
+  });
+
+  const open = service.openCdp();
+  const cdp = await open;
+
+  const lock = cdp.lockEth(0.1);
+  await Promise.all([lock, mineBlocks(service)]);
+
+  const draw = cdp.drawDai(13);
+  await Promise.all([txMgr.confirm(draw), mineBlocks(service)]);
+  
+  // set price to make cdp unsafe
+  await priceService.setEthPrice(0.01);
+
+  const bite = cdp.bite();
+  log('bite id:', uniqueId(bite));
+
+  const biteHandlers = makeHandlers('bite');
+
+  txMgr.listen(bite, biteHandlers);
+  
+  await bite;
+
+  expect(biteHandlers.pending).toBeCalled();
+  expect(biteHandlers.mined).toBeCalled();
+
+  // set price back to 400
+  await priceService.setEthPrice(400);
```

Comment:
This has to happen regardless of test failure, so it should be in an `afterEach` block.